keylime/web_util.py

```diff
@@ -11,117 +10,195 @@
+
         if os.path.exists(ca_path):
             if logger:
                 logger.info("Existing CA certificate found in %s, not generating a new one", tls_dir)
-        else:
+            return tls_dir
+
+        for option in options:
+            value = config.get(component, option)
+            if value != "default":
+                raise Exception(f"To use tls_dir=generate, the following options must be set to 'default': {options}")
+
+        if logger:
+            logger.info("Generating new CA, keys, and certificates in %s", tls_dir)
+            logger.info("use keylime_ca -d %s to manage this CA", tls_dir)
+
+        if not os.path.exists(tls_dir):
+            os.makedirs(tls_dir, 0o700)
+
+        password = config.get(component, "server_key_password")
+        if not password:
+            password = None
+        elif password == "default":
             if logger:
-                logger.info("Generating a new CA in %s and a client certificate for connecting", tls_dir)
-                logger.info("use keylime_ca -d %s to manage this CA", tls_dir)
-            if not os.path.exists(tls_dir):
-                os.makedirs(tls_dir, 0o700)
-            if my_key_pw == "default":
-                if logger:
-                    logger.warning(
-                        "CAUTION: using default password for CA, please set private_key_pw to a strong password"
-                    )
-            ca_util.setpassword(my_key_pw)
-            ca_util.cmd_init(tls_dir)
-            ca_util.cmd_mkcert(tls_dir, socket.gethostname())
-            ca_util.cmd_mkcert(tls_dir, "client")
-
-    if tls_dir == "CV":
-        if section != "registrar":
-            raise Exception(f"You only use the CV option to tls_dir for the registrar not {section}")
+                logger.warning("Using 'default' server_key_password option for %s", component)
+        else:
+            password = str(password)
+
+        ca_util.setpassword(password)
+        ca_util.cmd_init(tls_dir)
+        ca_util.cmd_mkcert(tls_dir, "server", password=password)
+
+        if component == "verifier":
+            password = config.get(component, "client_key_password")
+            if not password:
+                password = None
+            else:
+                password = str(password)
+
+            # The verifier also needs client key/certificate to access the agent
+            ca_util.cmd_mkcert(tls_dir, "client", password=password)
+
+    elif tls_dir == "default":
+        # Use the keys/certificates generated for the verifier
         tls_dir = os.path.abspath(os.path.join(config.WORK_DIR, "cv_ca"))
         if not os.path.exists(os.path.join(tls_dir, "cacert.crt")):
             raise Exception(
                 "It appears that the verifier has not yet created a CA and certificates, please run the verifier first"
             )
+    else:
+        # if it is relative path, convert to absolute in WORK_DIR
+        if tls_dir[0] != "/":
```

Comment:
```suggestion
        if not os.path.isabs(tls_dir):
```

Reply:
also not needed? (see previous instance)